<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Benvenuti nel Sito di Bioinformatica</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">BIOINFORMATICS</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="analisi_genomica.html">Analisi Genomica</a>
</li>
<li>
  <a href="visualizzazioni_dati.html">Visualizzazioni Dati</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/mauriziomurano/bioinformaticswebsite">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Benvenuti nel Sito di Bioinformatica</h1>

</div>


<div id="esplorando-la-bioinformatica-con-r" class="section level1">
<h1>Esplorando la Bioinformatica con R</h1>
<p>Benvenuti nel sito dedicato alla bioinformatica, dove esploreremo
analisi di dati biologici complessi utilizzando il potente linguaggio di
programmazione R.</p>
<p>Qui troverete risorse, tutorial ed esempi pratici su diverse aree
della bioinformatica, dall’analisi genomica alla visualizzazione di dati
complessi.</p>
<hr />
<div id="connettersi-al-database-ensembl-biomart"
class="section level2">
<h2>Connettersi al database Ensembl BioMart</h2>
<p>Cerchiamo la categoria (MART) dei Database Biomart</p>
<pre class="r"><code>library(biomaRt)
listMarts()</code></pre>
<pre><code>##                biomart                version
## 1 ENSEMBL_MART_ENSEMBL      Ensembl Genes 114
## 2   ENSEMBL_MART_MOUSE      Mouse strains 114
## 3     ENSEMBL_MART_SNP  Ensembl Variation 114
## 4 ENSEMBL_MART_FUNCGEN Ensembl Regulation 114</code></pre>
<p>Connessione al server principale di Ensembl BioMart tramite il
comando useMart(“ensembl”):</p>
<pre class="r"><code>A &lt;- useMart(&quot;ensembl&quot;)

listDatasets(A)[1:5,1:2]</code></pre>
<pre><code>##                        dataset                           description
## 1 abrachyrhynchus_gene_ensembl Pink-footed goose genes (ASM259213v1)
## 2     acalliptera_gene_ensembl      Eastern happy genes (fAstCal1.3)
## 3   acarolinensis_gene_ensembl       Green anole genes (AnoCar2.0v2)
## 4    acchrysaetos_gene_ensembl       Golden eagle genes (bAquChr1.2)
## 5    acitrinellus_gene_ensembl        Midas cichlid genes (Midas_v5)</code></pre>
<pre class="r"><code>str(A)</code></pre>
<pre><code>## Formal class &#39;Mart&#39; [package &quot;biomaRt&quot;] with 8 slots
##   ..@ biomart    : chr &quot;ENSEMBL_MART_ENSEMBL&quot;
##   ..@ host       : chr &quot;https://www.ensembl.org:443/biomart/martservice&quot;
##   ..@ vschema    : chr &quot;default&quot;
##   ..@ version    : chr &quot;&quot;
##   ..@ dataset    : chr &quot;&quot;
##   ..@ filters    :&#39;data.frame&#39;:  0 obs. of  0 variables
## Formal class &#39;data.frame&#39; [package &quot;methods&quot;] with 4 slots
##   .. .. ..@ .Data    : list()
##   .. .. ..@ names    : chr(0) 
##   .. .. ..@ row.names: int(0) 
##   .. .. ..@ .S3Class : chr &quot;data.frame&quot;
##   ..@ attributes :&#39;data.frame&#39;:  0 obs. of  0 variables
## Formal class &#39;data.frame&#39; [package &quot;methods&quot;] with 4 slots
##   .. .. ..@ .Data    : list()
##   .. .. ..@ names    : chr(0) 
##   .. .. ..@ row.names: int(0) 
##   .. .. ..@ .S3Class : chr &quot;data.frame&quot;
##   ..@ http_config: list()</code></pre>
<p>Greppiamo tutti i dataset che contengono la parola “sapiens”</p>
<pre class="r"><code>listDatasets(A)[grep(&quot;sapiens&quot;, listDatasets(A)$dataset), ]</code></pre>
<pre><code>##                  dataset              description    version
## 80 hsapiens_gene_ensembl Human genes (GRCh38.p14) GRCh38.p14</code></pre>
<p>Selezioniamo il dataset specifico per il genoma umano, cioè la specie
“Homo sapiens genes (GRCh38.p5)”</p>
<pre class="r"><code>B &lt;- useDataset(&quot;hsapiens_gene_ensembl&quot;, mart = A)

str(B)</code></pre>
<pre><code>## Formal class &#39;Mart&#39; [package &quot;biomaRt&quot;] with 8 slots
##   ..@ biomart    : chr &quot;ENSEMBL_MART_ENSEMBL&quot;
##   ..@ host       : chr &quot;https://www.ensembl.org:443/biomart/martservice&quot;
##   ..@ vschema    : chr &quot;default&quot;
##   ..@ version    : chr &quot;&quot;
##   ..@ dataset    : chr &quot;hsapiens_gene_ensembl&quot;
##   ..@ filters    :&#39;data.frame&#39;:  441 obs. of  9 variables:
##   .. ..$ name           : chr [1:441] &quot;chromosome_name&quot; &quot;start&quot; &quot;end&quot; &quot;band_start&quot; ...
##   .. ..$ description    : chr [1:441] &quot;Chromosome/scaffold name&quot; &quot;Start&quot; &quot;End&quot; &quot;Band Start&quot; ...
##   .. ..$ options        : chr [1:441] &quot;[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,GL000008.2,GL000009.2,GL000194.1,GL000195.1,GL000205.&quot;| __truncated__ &quot;[]&quot; &quot;[]&quot; &quot;[]&quot; ...
##   .. ..$ fullDescription: chr [1:441] &quot;&quot; &quot;Determine which base pair on the specified chromosome/scaffold to begin range&quot; &quot;Determine which base pair on the specified chromosome/scaffold to end range&quot; &quot;&quot; ...
##   .. ..$ filters        : chr [1:441] &quot;filters&quot; &quot;filters&quot; &quot;filters&quot; &quot;filters&quot; ...
##   .. ..$ type           : chr [1:441] &quot;text&quot; &quot;text&quot; &quot;text&quot; &quot;text&quot; ...
##   .. ..$ operation      : chr [1:441] &quot;=&quot; &quot;&gt;=&quot; &quot;&lt;=&quot; &quot;=&quot; ...
##   .. ..$ NA             : chr [1:441] &quot;hsapiens_gene_ensembl__gene__main&quot; &quot;hsapiens_gene_ensembl__gene__main&quot; &quot;hsapiens_gene_ensembl__gene__main&quot; &quot;pointer dataset&quot; ...
##   .. ..$ NA             : chr [1:441] &quot;name_1059&quot; &quot;seq_region_end_1020&quot; &quot;seq_region_start_1020&quot; &quot;band_1027&quot; ...
##   ..@ attributes :&#39;data.frame&#39;:  3170 obs. of  7 variables:
##   .. ..$ name           : chr [1:3170] &quot;ensembl_gene_id&quot; &quot;ensembl_gene_id_version&quot; &quot;ensembl_transcript_id&quot; &quot;ensembl_transcript_id_version&quot; ...
##   .. ..$ description    : chr [1:3170] &quot;Gene stable ID&quot; &quot;Gene stable ID version&quot; &quot;Transcript stable ID&quot; &quot;Transcript stable ID version&quot; ...
##   .. ..$ fullDescription: chr [1:3170] &quot;Stable ID of the Gene&quot; &quot;Versionned stable ID of the Gene&quot; &quot;Stable ID of the Transcript&quot; &quot;Versionned stable ID of the Transcript&quot; ...
##   .. ..$ page           : chr [1:3170] &quot;feature_page&quot; &quot;feature_page&quot; &quot;feature_page&quot; &quot;feature_page&quot; ...
##   .. ..$ NA             : chr [1:3170] &quot;html,txt,csv,tsv,xls&quot; &quot;html,txt,csv,tsv,xls&quot; &quot;html,txt,csv,tsv,xls&quot; &quot;html,txt,csv,tsv,xls&quot; ...
##   .. ..$ NA             : chr [1:3170] &quot;hsapiens_gene_ensembl__gene__main&quot; &quot;hsapiens_gene_ensembl__gene__main&quot; &quot;hsapiens_gene_ensembl__transcript__main&quot; &quot;hsapiens_gene_ensembl__transcript__main&quot; ...
##   .. ..$ NA             : chr [1:3170] &quot;stable_id_1023&quot; &quot;gene__main_stable_id_version&quot; &quot;stable_id_1066&quot; &quot;transcript__main_stable_id_version&quot; ...
##   ..@ http_config: list()</code></pre>
<p>Lista degli attributi del Dataset Genoma Umano (ne riportiamo solo 5
per brevità):</p>
<pre class="r"><code>B@attributes[1:5,1:2]</code></pre>
<pre><code>##                            name                  description
## 1               ensembl_gene_id               Gene stable ID
## 2       ensembl_gene_id_version       Gene stable ID version
## 3         ensembl_transcript_id         Transcript stable ID
## 4 ensembl_transcript_id_version Transcript stable ID version
## 5            ensembl_peptide_id            Protein stable ID</code></pre>
<p>Chiediamo la lista dei cromosomi. I primi valori, come “1”, “2”, …,
“22”, sono i cromosomi numerici standard.</p>
<pre class="r"><code>CHR &lt;- getBM(
  attributes = &quot;chromosome_name&quot;,
  mart = B
)

CHR[1:10,]</code></pre>
<pre><code>##  [1] &quot;1&quot;  &quot;10&quot; &quot;11&quot; &quot;12&quot; &quot;13&quot; &quot;14&quot; &quot;15&quot; &quot;16&quot; &quot;17&quot; &quot;18&quot;</code></pre>
<p>Di tutto il genoma umano filtriamo il solo cromosoma 21 con i suoi
attributi: “Ensembl Gene ID”, “Ensembl Transcript ID”, “Associated Gene
Name” e “Description”.</p>
<pre class="r"><code>CHR21 &lt;- getBM(
  attributes = c(&quot;ensembl_gene_id&quot;,
    &quot;ensembl_transcript_id&quot;,
    &quot;external_gene_name&quot;,
    &quot;description&quot;),
  filters = &quot;chromosome_name&quot;,
  values = &quot;21&quot;,
  mart = B
)

CHR21[1:5,]</code></pre>
<pre><code>##   ensembl_gene_id ensembl_transcript_id external_gene_name
## 1 ENSG00000286057       ENST00000652328                   
## 2 ENSG00000286146       ENST00000652246                   
## 3 ENSG00000278678       ENST00000616920            CYCSP41
## 4 ENSG00000279647       ENST00000624595                   
## 5 ENSG00000280372       ENST00000624662                   
##                                                    description
## 1                                                          TEC
## 2                                                          TEC
## 3        CYCS pseudogene 41 [Source:HGNC Symbol;Acc:HGNC:2580]
## 4 clustered mitochondria (cluA/CLU1) homolog (CLUH) pseudogene
## 5                                                          TEC</code></pre>
<pre class="r"><code>dim(CHR21)</code></pre>
<pre><code>## [1] 5645    4</code></pre>
<p>Osserviamo che ogni gene può generare più di un trascritti:</p>
<pre class="r"><code>TRAS_CHR21 &lt;- CHR21[order(CHR21$ensembl_gene_id), ][1:10,1:2]
TRAS_CHR21</code></pre>
<pre><code>##     ensembl_gene_id ensembl_transcript_id
## 576 ENSG00000141956       ENST00000470586
## 577 ENSG00000141956       ENST00000486812
## 578 ENSG00000141956       ENST00000447016
## 579 ENSG00000141956       ENST00000398548
## 580 ENSG00000141956       ENST00000422911
## 581 ENSG00000141956       ENST00000441787
## 582 ENSG00000141956       ENST00000449395
## 583 ENSG00000141956       ENST00000269844
## 584 ENSG00000141956       ENST00000465955
## 585 ENSG00000141956       ENST00000477633</code></pre>
</div>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
