<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Microarray Analysis</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">BIOINFORMATICS</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="PRIMA.html">PRIMA</a>
</li>
<li>
  <a href="SECONDA.html">SECONDA</a>
</li>
<li>
  <a href="ALLINEAMENTO.html">ALLINEAMENTO</a>
</li>
<li>
  <a href="Microarray Analysis.html">Microarray Analysis</a>
</li>
<li>
  <a href="BANCHE.html">BANCHE</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/mauriziomurano/bioinformaticswebsite">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Microarray Analysis</h1>

</div>


<div
id="matrice-id-delle-sonde---id-dei-campioni-e-selezione-dei-campioni-primi-12"
class="section level3">
<h3>Matrice “ID delle sonde” - “ID dei campioni” e selezione dei
campioni (Primi 12)</h3>
<p><img src="img1.png" width="180" /></p>
<pre class="r"><code>A &lt;- getGEO(&quot;GSE76999&quot;, GSEMatrix = TRUE, getGPL = FALSE)</code></pre>
<pre><code>## Found 1 file(s)</code></pre>
<pre><code>## GSE76999_series_matrix.txt.gz</code></pre>
<pre class="r"><code>B &lt;- A[[1]]

varLabels(B)</code></pre>
<pre><code>##  [1] &quot;title&quot;                   &quot;geo_accession&quot;           &quot;status&quot;                 
##  [4] &quot;submission_date&quot;         &quot;last_update_date&quot;        &quot;type&quot;                   
##  [7] &quot;channel_count&quot;           &quot;source_name_ch1&quot;         &quot;organism_ch1&quot;           
## [10] &quot;characteristics_ch1&quot;     &quot;characteristics_ch1.1&quot;   &quot;characteristics_ch1.2&quot;  
## [13] &quot;treatment_protocol_ch1&quot;  &quot;growth_protocol_ch1&quot;     &quot;molecule_ch1&quot;           
## [16] &quot;extract_protocol_ch1&quot;    &quot;label_ch1&quot;               &quot;label_protocol_ch1&quot;     
## [19] &quot;taxid_ch1&quot;               &quot;hyb_protocol&quot;            &quot;scan_protocol&quot;          
## [22] &quot;description&quot;             &quot;data_processing&quot;         &quot;data_processing.1&quot;      
## [25] &quot;data_processing.2&quot;       &quot;platform_id&quot;             &quot;contact_name&quot;           
## [28] &quot;contact_email&quot;           &quot;contact_department&quot;      &quot;contact_institute&quot;      
## [31] &quot;contact_address&quot;         &quot;contact_city&quot;            &quot;contact_zip/postal_code&quot;
## [34] &quot;contact_country&quot;         &quot;supplementary_file&quot;      &quot;data_row_count&quot;         
## [37] &quot;relation&quot;                &quot;relation.1&quot;              &quot;age:ch1&quot;                
## [40] &quot;strain:ch1&quot;              &quot;tissue:ch1&quot;</code></pre>
<pre class="r"><code>B$organism_ch1</code></pre>
<pre><code>##  [1] &quot;Mus musculus&quot; &quot;Mus musculus&quot; &quot;Mus musculus&quot; &quot;Mus musculus&quot; &quot;Mus musculus&quot; &quot;Mus musculus&quot;
##  [7] &quot;Mus musculus&quot; &quot;Mus musculus&quot; &quot;Mus musculus&quot; &quot;Mus musculus&quot; &quot;Mus musculus&quot; &quot;Mus musculus&quot;
## [13] &quot;Mus musculus&quot; &quot;Mus musculus&quot; &quot;Mus musculus&quot; &quot;Mus musculus&quot; &quot;Mus musculus&quot; &quot;Mus musculus&quot;
## [19] &quot;Mus musculus&quot; &quot;Mus musculus&quot; &quot;Mus musculus&quot; &quot;Mus musculus&quot; &quot;Mus musculus&quot; &quot;Mus musculus&quot;
## [25] &quot;Mus musculus&quot; &quot;Mus musculus&quot; &quot;Mus musculus&quot; &quot;Mus musculus&quot; &quot;Mus musculus&quot; &quot;Mus musculus&quot;
## [31] &quot;Mus musculus&quot; &quot;Mus musculus&quot; &quot;Mus musculus&quot; &quot;Mus musculus&quot; &quot;Mus musculus&quot; &quot;Mus musculus&quot;</code></pre>
<pre class="r"><code>B$title  # Analogo a B$description</code></pre>
<pre><code>##  [1] &quot;Monocyte extracted from adult (wk6-12) Bone Marrow, biological replicate 1&quot;                                     
##  [2] &quot;Monocyte extracted from adult (wk6-12) Bone Marrow, biological replicate 2&quot;                                     
##  [3] &quot;Monocyte extracted from adult (wk6-12) Bone Marrow, biological replicate 3&quot;                                     
##  [4] &quot;Monocyte extracted from adult (wk6-12) Bone Marrow, biological replicate 4&quot;                                     
##  [5] &quot;Monocyte extracted from E15.5 Fetal Liver, biological replicate 1&quot;                                              
##  [6] &quot;Monocyte extracted from E15.5 Fetal Liver, biological replicate 2&quot;                                              
##  [7] &quot;Monocyte extracted from E15.5 Fetal Liver, biological replicate 3&quot;                                              
##  [8] &quot;Monocyte extracted from E15.5 Fetal Liver, biological replicate 4&quot;                                              
##  [9] &quot;Macrophage extracted from E12.5 Yolk Sac, biological replicate 1&quot;                                               
## [10] &quot;Macrophage extracted from E12.5 Yolk Sac, biological replicate 2&quot;                                               
## [11] &quot;Macrophage extracted from E12.5 Yolk Sac, biological replicate 3&quot;                                               
## [12] &quot;Macrophage extracted from E12.5 Yolk Sac, biological replicate 4&quot;                                               
## [13] &quot;Alveolar Macrophage extracted from adult (wk6-12) mice via Bronchoalveolar lavage, biological replicate 1&quot;      
## [14] &quot;Alveolar Macrophage extracted from adult (wk6-12) mice via Bronchoalveolar lavage, biological replicate 2&quot;      
## [15] &quot;Alveolar Macrophage extracted from adult (wk6-12) mice via Bronchoalveolar lavage, biological replicate 3&quot;      
## [16] &quot;Alveolar Macrophage extracted from adult (wk6-12) mice via Bronchoalveolar lavage, biological replicate 4&quot;      
## [17] &quot;Alveolar Macrophage derived from transferred Bone Marrow Monocytes, 6 wks post-transfer, biological replicate 1&quot;
## [18] &quot;Alveolar Macrophage derived from transferred Bone Marrow Monocytes, 6 wks post-transfer, biological replicate 2&quot;
## [19] &quot;Alveolar Macrophage derived from transferred Bone Marrow Monocytes, 6 wks post-transfer, biological replicate 3&quot;
## [20] &quot;Alveolar Macrophage derived from transferred Bone Marrow Monocytes, 6 wks post-transfer, biological replicate 4&quot;
## [21] &quot;Alveolar Macrophage derived from transferred Fetal Liver Monocytes, 6 wks post-transfer, biological replicate 1&quot;
## [22] &quot;Alveolar Macrophage derived from transferred Fetal Liver Monocytes, 6 wks post-transfer, biological replicate 2&quot;
## [23] &quot;Alveolar Macrophage derived from transferred Fetal Liver Monocytes, 6 wks post-transfer, biological replicate 3&quot;
## [24] &quot;Alveolar Macrophage derived from transferred Fetal Liver Monocytes, 6 wks post-transfer, biological replicate 4&quot;
## [25] &quot;Alveolar Macrophage derived from transferred Yolk Sac Macrophages, 6 wks post-transfer, biological replicate 1&quot; 
## [26] &quot;Alveolar Macrophage derived from transferred Yolk Sac Macrophages, 6 wks post-transfer, biological replicate 2&quot; 
## [27] &quot;Alveolar Macrophage derived from transferred Yolk Sac Macrophages, 6 wks post-transfer, biological replicate 3&quot; 
## [28] &quot;Alveolar Macrophage derived from transferred Yolk Sac Macrophages, 6 wks post-transfer, biological replicate 4&quot; 
## [29] &quot;Alveolar Macrophage derived from transferred Alveolar macrophages, 6 wks post-transfer, biological replicate 1&quot; 
## [30] &quot;Alveolar Macrophage derived from transferred Alveolar macrophages, 6 wks post-transfer, biological replicate 2&quot; 
## [31] &quot;Alveolar Macrophage derived from transferred Alveolar macrophages, 6 wks post-transfer, biological replicate 3&quot; 
## [32] &quot;Alveolar Macrophage derived from transferred Alveolar macrophages, 6 wks post-transfer, biological replicate 4&quot; 
## [33] &quot;Alveolar Macrophage extracted from 6 wk old wild type mice via Bronchoalveolar lavage, biological replicate 1&quot;  
## [34] &quot;Alveolar Macrophage extracted from 6 wk old wild type mice via Bronchoalveolar lavage, biological replicate 2&quot;  
## [35] &quot;Alveolar Macrophage extracted from 6 wk old wild type mice via Bronchoalveolar lavage, biological replicate 3&quot;  
## [36] &quot;Alveolar Macrophage extracted from 6 wk old wild type mice via Bronchoalveolar lavage, biological replicate 4&quot;</code></pre>
<pre class="r"><code>C &lt;- exprs(B)

C[1:5,1:12]</code></pre>
<pre><code>##          GSM2042244 GSM2042245 GSM2042246 GSM2042247 GSM2042248 GSM2042249 GSM2042250 GSM2042251
## 10338001  12.883736  12.710357  12.808904  12.728285  12.670743  12.582576  12.728665  12.701661
## 10338002   5.478638   4.836992   5.354476   5.311100   5.128454   5.158269   5.576625   5.174163
## 10338003  11.565683  11.007533  11.348052  11.232528  10.989176  11.017759  11.300946  11.105487
## 10338004   9.784679   9.218110   7.900875   9.508223   7.373025   7.647630   9.188661   9.042846
## 10338005   2.179659   1.997411   2.215015   2.134441   2.062177   2.174181   2.091476   2.102721
##          GSM2042252 GSM2042253 GSM2042254 GSM2042255
## 10338001  12.845415  12.763705  12.760986  12.609271
## 10338002   4.967542   5.210838   5.116310   4.890626
## 10338003  11.282599  11.216054  11.144170  10.939133
## 10338004   7.527401   9.193056   7.072794   9.433758
## 10338005   2.073256   2.095818   2.140202   2.067220</code></pre>
<pre class="r"><code>boxplot(C[1:2,1:12], las=2, cex.axis = 0.7)</code></pre>
<p><img src="Microarray-Analysis_files/figure-html/unnamed-chunk-3-1.png" width="672" /></p>
</div>
<div id="differential-expression-analysis-by-limma"
class="section level3">
<h3>Differential Expression Analysis by Limma</h3>
<p>Adesso cerchiamo quali geni sono espressi con delle differenze tra le
tre categorie di esperimento:</p>
<p>Bone Marrow</p>
<p>Fetal Liver</p>
<p>Yolk Sac</p>
<pre class="r"><code># Trasformiamo i valori in scala logaritmica (ex = C)
D &lt;- log2(C)

labels &lt;- c(&quot;BM_MOs&quot;,&quot;FL_MOs&quot;,&quot;YS_Macs&quot;)
sel_groups &lt;- factor(rep(labels, each=4))


# Consideriamo E come il sottoinsieme dell&#39;oggetto B ridotto a 12 colonne
E &lt;- B[,1:12]

E$description &lt;- sel_groups

design &lt;- model.matrix(~ description + 0, E)

colnames(design) &lt;- levels(sel_groups)


fit &lt;- lmFit(E, design)


# Build comparison and compute the satistics: Dici a limma esattamente quali gruppi vuoi confrontare tra loro (es. BM_MOs contro FL_MOs).
cont.matrix &lt;- makeContrasts(BM_MOs-FL_MOs, BM_MOs-YS_Macs, FL_MOs-YS_Macs, levels=design)
# Apply the contrast matrix to the linear model - Esegue i confronti
fit2 &lt;- contrasts.fit(fit, cont.matrix)
# Generate the statistics (proportion -&gt; assumed proportion of genes which are differentially expressed): Migliora le statistiche. eBayes è un passo fondamentale di limma che usa una tecnica statistica per &quot;prestare&quot; informazioni tra i geni, migliorando l&#39;affidabilità delle stime di variabilità. Questo è cruciale per ottenere valori P più precisi, specialmente quando si hanno pochi campioni.
fit2 &lt;- eBayes(fit2, proportion = 0.05)

# Extract the sig. DE genesCosa fa? Estrae i risultati finali. La funzione topTable() prende il modello fit2 e lo usa per creare una tabella con i  geni che mostrano una differenza di espressione statisticamente significativa. Usa il parametro coef per specificare quale confronto vuoi vedere (ad esempio, coef=1 per BM_MOs-FL_MOs).
# coef=1  BM_MOs-FL_MOs
# coef=2  BM_MOs-YS_Macs
# coef=3  FL_MOs-YS_Macs
de_genes_BM_MOs_FL_MOs &lt;- topTable(fit2, coef=1, adjust=&quot;fdr&quot;, p.value=0.05, number=Inf)
de_genes_BM_MOs_YS_Macs &lt;- topTable(fit2, coef=2, adjust=&quot;fdr&quot;, p.value=0.05, number=Inf)
de_genes_FL_MOs_YS_Macs &lt;- topTable(fit2, coef=3, adjust=&quot;fdr&quot;, p.value=0.05, number=Inf)

head(de_genes_BM_MOs_FL_MOs) # show top de genes in BM_MO vs FL_MO comparirson. </code></pre>
<pre><code>##              logFC  AveExpr         t      P.Value    adj.P.Val        B
## 10435497 -8.689962 5.561077 -46.92658 1.881934e-14 6.691403e-10 16.30296
## 10501020  7.496752 6.800520  36.32079 3.472725e-13 6.173810e-09 15.72408
## 10387568  2.975392 6.801732  26.57701 1.193418e-11 1.414439e-07 14.64075
## 10608637 -8.113364 8.242008 -25.58598 1.831783e-11 1.628272e-07 14.47541
## 10400143  4.879170 6.266135  24.84392 2.551738e-11 1.814592e-07 14.34200
## 10435504 -6.364942 4.983137 -24.35867 3.185952e-11 1.887995e-07 14.24994</code></pre>
<pre class="r"><code># Get all the DE genes regardless the comparisons
all_de_genes &lt;- topTable(fit2, adjust=&quot;fdr&quot;, p.value=0.05, number=Inf)
all_genes_BM_MOs_FL_MOs &lt;- topTable(fit2, coef=1, adjust=&quot;fdr&quot;, p.value=1, number=Inf)
# These tables will be used later

# Output in an excel file
wb &lt;- createWorkbook()
# Create each sheet
addWorksheet(wb, &quot;de_genes_BM_MOs_FL_MOs&quot;)
addWorksheet(wb, &quot;de_genes_BM_MOs_YS_Macs&quot;)
addWorksheet(wb, &quot;de_genes_FL_MOs_YS_Macs&quot;)
# Write the data into each sheet
writeData(wb, &quot;de_genes_BM_MOs_FL_MOs&quot;, de_genes_BM_MOs_FL_MOs, rowNames = FALSE)
writeData(wb, &quot;de_genes_BM_MOs_YS_Macs&quot;, de_genes_BM_MOs_YS_Macs, rowNames = FALSE)
writeData(wb, &quot;de_genes_FL_MOs_YS_Macs&quot;, de_genes_FL_MOs_YS_Macs, rowNames = FALSE)
saveWorkbook(wb, &quot;DE_genes_GSE76999.xlsx&quot;, overwrite = TRUE)</code></pre>
<pre class="r"><code>columns(mogene10sttranscriptcluster.db)</code></pre>
<pre><code>##  [1] &quot;ACCNUM&quot;       &quot;ALIAS&quot;        &quot;ENSEMBL&quot;      &quot;ENSEMBLPROT&quot;  &quot;ENSEMBLTRANS&quot; &quot;ENTREZID&quot;    
##  [7] &quot;ENZYME&quot;       &quot;EVIDENCE&quot;     &quot;EVIDENCEALL&quot;  &quot;GENENAME&quot;     &quot;GENETYPE&quot;     &quot;GO&quot;          
## [13] &quot;GOALL&quot;        &quot;IPI&quot;          &quot;MGI&quot;          &quot;ONTOLOGY&quot;     &quot;ONTOLOGYALL&quot;  &quot;PATH&quot;        
## [19] &quot;PFAM&quot;         &quot;PMID&quot;         &quot;PROBEID&quot;      &quot;PROSITE&quot;      &quot;REFSEQ&quot;       &quot;SYMBOL&quot;      
## [25] &quot;UNIPROT&quot;</code></pre>
<pre class="r"><code>ID &lt;- featureNames(E)

tmp &lt;- select(mogene10sttranscriptcluster.db, ID, c(&quot;SYMBOL&quot;, &quot;ENTREZID&quot;))</code></pre>
<pre><code>## &#39;select()&#39; returned 1:many mapping between keys and columns</code></pre>
<pre class="r"><code>tmp &lt;- na.omit(tmp)

all_genes_BM_MOs_FL_MOs$ENTREZID &lt;- tmp$ENTREZID[match(rownames(all_genes_BM_MOs_FL_MOs), tmp$PROBEID)]
all_de_genes$symbol &lt;- tmp$SYMBOL[match(rownames(all_de_genes), tmp$PROBEID)]
all_de_genes &lt;- na.omit(all_de_genes)
de_genes_BM_MOs_FL_MOs$ENTREZID &lt;- tmp$ENTREZID[match(rownames(de_genes_BM_MOs_FL_MOs), tmp$PROBEID)]
#de_genes_BM_MOs_FL_MOs$symbol &lt;- na.omit(tmp$ENTREZID[match(rownames(de_genes_BM_MOs_FL_MOs), tmp$PROBEID)])
de_genes_BM_MOs_YS_Macs$ENTREZID &lt;- tmp$ENTREZID[match(rownames(de_genes_BM_MOs_YS_Macs), tmp$PROBEID)]
#de_genes_BM_MOs_YS_Macs$symbol &lt;- na.omit(tmp$ENTREZID[match(rownames(de_genes_BM_MOs_YS_Macs), tmp$PROBEID)])
de_genes_FL_MOs_YS_Macs$ENTREZID &lt;- tmp$ENTREZID[match(rownames(de_genes_FL_MOs_YS_Macs), tmp$PROBEID)]
#de_genes_FL_MOs_YS_Macs$symbol &lt;- na.omit(tmp$ENTREZID[match(rownames(de_genes_FL_MOs_YS_Macs), tmp$PROBEID)])

ma &lt;- fit2[,&quot;BM_MOs - FL_MOs&quot;]
limma::plotMA(ma)</code></pre>
<p><img src="Microarray-Analysis_files/figure-html/unnamed-chunk-5-1.png" width="672" /></p>
<p>Cosa puoi dedurre dal tuo grafico?</p>
<p>La forma a “ventaglio” o “imbuto” è tipica e attesa. I geni a bassa
espressione (a sinistra del grafico) mostrano una maggiore variabilità
nel log-fold change, mentre i geni ad alta espressione (a destra) sono
molto più stabili e raggruppati intorno allo zero. Questo è un buon
segno e indica che i dati sono stati normalizzati correttamente.</p>
<p>I punti più lontani dalla linea orizzontale dello 0 sull’asse Y
rappresentano i geni con la più grande differenza di espressione tra
BM-MOs e FL-MOs. Molti di questi punti (che rappresentano i geni
differenzialmente espressi) si concentrano nelle regioni a più alta
espressione, come si può notare dalla coda del grafico.</p>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
